src/_pytest/_code/code.py

```diff
@@ -421,14 +421,14 @@ class ExceptionInfo(Generic[_E]):
 
     _assert_start_repr = "AssertionError('assert "
 
-    _excinfo = attr.ib(type=Optional[Tuple["Type[_E]", "_E", TracebackType]])
+    _excinfo = attr.ib(type=Optional[Tuple[Type["_E"], "_E", TracebackType]])
```

Comment:
Does `_E` needs to be quoted here?

Reply:
```suggestion
    _excinfo = attr.ib(type=Optional[Tuple[Type[_E], _E, TracebackType]])
```

I don't think so

Reply:
can't this be an actual type annotation now?

```suggestion
    _excinfo: Optional[Tuple[Type[_E], _E, TracebackType] = attr.ib()
```

or even with `auto_attribs=True`?

Reply:
Agreed

Reply:
there's a bug in mypy here fwiw